### PR TITLE
Moved internal.rb to be under the ruby_lsp directory

### PR DESCRIPTION
**Pull request by Smittttty**
### Motivation

PaymentService has an engine thats named [Internal](https://github.com/Shopify/payment-service/blob/c5d067edabe88f5be18a740e8dd5a128d8c30c0a/engines/internal/internal.gemspec#L3-L19). With the latest version of `ruby-lsp`, PaymentService fails to load the Internal component. [Related Buildkite](https://buildkite.com/shopify/shopify-slash-payment-service-ci/builds/35187#018162b7-1ce7-4855-b3be-ebe3b9a6e9f3).

https://github.com/Shopify/ruby-lsp/pull/119/commits/586eccfa5f6de1640a58b9c5f4d364db6a314aca was the only change that seemed like it could have caused this. Renaming the `internal.rb` file resolves the issue, as does moving the file to not be in the top-level `/lib` folder.

### Implementation

Moved `internal.rb` to be under the `ruby_lsp` directory.

### CODE REVIEW

**Comment by st0012** [comment]
I think maybe the problem is that there shouldn't be any non-`ruby-lsp.rb` file located at the top-level `/lib`. Even if we changed it to `requires.rb`, it can still be loaded with `require "requires"`, which may be some other gems' name. So the better solution to me is to place the `internal.rb` under the `ruby-lsp` namespace.

**Comment by paracycle** [comment]
Yeah 100% agreed with what @st0012 wrote above. We should be moving the require to inside the `ruby-lsp` folder.

**Review by paracycle — CHANGES_REQUESTED** [review]

File: lib/ruby_lsp/internal.rb [hunk 1]
```diff
@@ -4,4 +4,4 @@
 require "sorbet-runtime"
 require "syntax_tree"
 require "ruby-lsp"
-require "ruby_lsp/cli"
+require "cli"
```
[the remark this answers is not in the document]
Comment (paracycle) [reply lib/ruby_lsp/internal.rb:7]: As evidenced by the test failure: https://github.com/Shopify/ruby-lsp/runs/6889307468?check_suite_focus=true#step:7:27

**Review by paracycle — APPROVED** [review]
Thanks and sorry that this broke your workflow!

**Comment by Smittttty** [comment]
> Thanks and sorry that this broke your workflow!

No problem, thanks for the quick review! ❤️

**Pull request merged**